Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.mm

```diff
@@ -294,14 +312,38 @@
 
 void ScrollingTreeScrollingNodeDelegateMac::releaseReferencesToScrollerImpsOnTheMainThread()
 {
-    if (m_verticalScrollerImp || m_horizontalScrollerImp) {
+    if (m_scrollerPair->verticalScroller().scrollerImp() || m_scrollerPair->horizontalScroller().scrollerImp()) {
         // FIXME: This is a workaround in place for the time being since NSScrollerImps cannot be deallocated
         // on a non-main thread. rdar://problem/24535055
-        WTF::callOnMainThread([verticalScrollerImp = WTFMove(m_verticalScrollerImp), horizontalScrollerImp = WTFMove(m_horizontalScrollerImp)] {
+        WTF::callOnMainThread([verticalScrollerImp = m_scrollerPair->verticalScroller().scrollerImp(), horizontalScrollerImp = m_scrollerPair->horizontalScroller().scrollerImp()] {
```

Comment:
This needs to "take" the scrollerImps (they need to be nulled out on the pair), so that the last ref goes away on the main thread. Maybe ScrollerPairMac itself can take care of this bounce to the main thread.

Reply:
Fixed.